pkg/helper/client.go

```diff
@@ -122,9 +122,10 @@ func UpdateCurrentState(client client.Client, nodeNetworkState *nmstatev1beta1.N
 
 	err = client.Status().Update(context.Background(), nodeNetworkState)
 	if err != nil {
-		// Request object not found, could have been deleted after reconcile request.
-		if !apierrors.IsNotFound(err) {
+		if apierrors.IsNotFound(err) {
 			return errors.Wrap(err, "Request object not found, could have been deleted after reconcile request")
+		} else {
+			return errors.Wrap(err, "Error updating nodeNetworkState")
```

Comment:
This will break conflict resolution, that's why is done this way.

Reply:
we check `!apierrors.IsNotFound(err)` (mind the **!**)  and if that condition is true, we
`return errors.Wrap(err, "Request object not found, could have been deleted after reconcile request")`.
But that is surely the only error that did not occur

Reply:
but are we wrapping this function on `RetryOnConflict` for this to work we cannot wrap errors.

Reply:
We are already wrapping the error though

Reply:
Maybe we don't want to return an error when the object is not found, in that case it's only the error message that isn't right